array-methods.js

```diff
@@ -102,20 +158,80 @@ var stateSums = null;
     round this number to the nearest 10th of a cent before moving on.
   )
  */
-var sumOfHighInterests = null;
+
+var sumOfHighInterests = 0;
+let filteredOtherArray = bankBalances;
+
+
+statesArray.forEach(currentState => {
+  filteredOtherArray = filteredOtherArray.filter(currentAccount => {
+    return (currentAccount.state !== currentState);
+  });
+});
+
+var otherStateSums = {};
+
+filteredOtherArray.forEach(currentAccount => {
+  let balance = parseFloat(currentAccount.amount);
+  let state = currentAccount.state;
+
+  if (!otherStateSums[state]) {
+    otherStateSums[state] = 0.0;
+  }
+
+  otherStateSums[state] = roundAndAdd(otherStateSums[state], balance);
+
+});
+
+sumOfHighInterests = Object.values(otherStateSums)
+  .reduce((sum, currentStateTotal) => {
+    let interest = parseFloat(currentStateTotal) * 0.189;
+    if (interest > 50000) {
+      sum = roundAndAdd(sum + interest);
+    }
+
+    return sum;
+
+  }, 0);
 
 /*
   set `lowerSumStates` to be an array of two letter state
   abbreviations of each state where the sum of amounts
   in the state is less than 1,000,000
  */
-var lowerSumStates = null;
+var lowerSumStates = [];
+let totalStateSums = {};
+
+bankBalances.forEach(currentAccount => {
```

Comment:
Dont rely so heavily on `.forEach`. Keep in-mind the `forEach` returns `undefined` making not useful for chaining.